buffer/src/main/java/io/netty/buffer/PoolArena.java

```diff
@@ -617,6 +617,29 @@ private static void appendPoolSubPages(StringBuilder buf, PoolSubpage<?>[] subpa
         }
     }
 
+    @Override
+    protected final void finalize() throws Throwable {
```

Comment:
![CRITICAL](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-critical.png) Do not override the Object.finalize() method. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://garage.netty.io/sonarqube/coding_rules#rule_key=squid%3AObjectFinalizeOverridenCheck)